diff --git a/ext/openssl/ossl_pkcs5.c b/ext/openssl/ossl_pkcs5.c
index ca02a18..007889f 100644
--- a/ext/openssl/ossl_pkcs5.c
+++ b/ext/openssl/ossl_pkcs5.c
@@ -27,7 +27,7 @@ ossl_pkcs5_pbkdf2_hmac(VALUE self, VALUE pass, VALUE salt, VALUE iter, VALUE key
 {
 #ifdef HAVE_PKCS5_PBKDF2_HMAC
     VALUE str;
-    const EVP_MD md;
+    const EVP_MD *md;
     int len = NUM2INT(keylen);
 
     StringValue(pass);
